R/zchunk_LA154.transportation_UCD.R

```diff
@@ -451,43 +452,43 @@ module_energy_LA154.transportation_UCD <- function(command, ...) {
-      ALL_ctry_var <- as.data.table(ALL_ctry_var)
-      ALL_ctry_var[, speed.x := if_else(is.na(speed.x), speed.y, speed.x)]
-      ALL_ctry_var[, speed.x := if_else(is.na(speed.x), 1, speed.x)]
-      ALL_ctry_var <- as_tibble(ALL_ctry_var)
-      #Separate out this by CORE and for SSPs
-      ALL_ctry_var_CORE <- ALL_ctry_var %>%
-        filter(sce=="CORE") %>%
-        select(-sce) %>%
-        select("UCD_technology","UCD_fuel", "UCD_sector", "rev.mode", "rev_size.class","mode","size.class", "year", "GCAM_region_ID","load factor","weight_EJ","intensity","non-fuel costs") %>%
-        rename(loadfactor_CORE = `load factor`, weight_EJ_core = weight_EJ, intensity_CORE =intensity, non_fuel_cost_core = `non-fuel costs`)
-
-
-      ALL_ctry_var %>%
+    #kbn 2020 Using data tables here instead of dplyr to increase processing time. Changes described in detail in comment with search string,kbn 2020-03-26.
+    ALL_ctry_var <- as.data.table(ALL_ctry_var)
+    ALL_ctry_var[, speed.x := if_else(is.na(speed.x), speed.y, speed.x)]
+    ALL_ctry_var[, speed.x := if_else(is.na(speed.x), 1, speed.x)]
+    ALL_ctry_var <- as_tibble(ALL_ctry_var)
+    #Separate out this by CORE and for SSPs
+    ALL_ctry_var_CORE <- ALL_ctry_var %>%
+      filter(sce=="CORE") %>%
+      select(-sce) %>%
+      select("UCD_technology","UCD_fuel", "UCD_sector", "rev.mode", "rev_size.class","mode","size.class", "year", "GCAM_region_ID","load factor","weight_EJ","intensity","non-fuel costs") %>%
+      rename(loadfactor_CORE = `load factor`, weight_EJ_core = weight_EJ, intensity_CORE =intensity, non_fuel_cost_core = `non-fuel costs`)
+
+
+    ALL_ctry_var %>%
       filter(sce != "CORE") %>%
       left_join_keep_first_only(ALL_ctry_var_CORE, by= c("UCD_technology","UCD_fuel", "UCD_sector", "mode", "size.class","rev.mode","rev_size.class", "year", "GCAM_region_ID")) %>%
       mutate(weight_EJ =if_else(is.na(weight_EJ),weight_EJ_core,weight_EJ),intensity =if_else(is.na(intensity),intensity_CORE,intensity), `load factor`=if_else(is.na(`load factor`),loadfactor_CORE,`load factor`), `non-fuel costs`=if_else(`non-fuel costs` == 0,non_fuel_cost_core,`non-fuel costs`)) %>%
       select(-loadfactor_CORE,-weight_EJ_core,-intensity_CORE,-non_fuel_cost_core) %>%
       filter(`non-fuel costs` != 0)-> ALL_ctry_var_SSPS
 
 
-      #kbn 2020 bind using rbindlist to increase processing speed. A separate PR submitted on github to functionalize this for future use.
-      list_for_bind=list ((ALL_ctry_var %>%  filter(sce=="CORE")), (ALL_ctry_var_SSPS) )
-      ALL_ctry_var <- rbindlist(list_for_bind, use.names=TRUE)
+    #kbn 2020 bind using rbindlist to increase processing speed. A separate PR submitted on github to functionalize this for future use.
+    list_for_bind=list ((ALL_ctry_var %>%  filter(sce=="CORE")), (ALL_ctry_var_SSPS) )
```

Comment:
I'm sure this isn't you @enlochner but can we change at least to `<-` instead of `=`? Ugh.